### ISSUE TITLE
(P1) MUST FIX: nxos providers are not working properly when transport=cli

### ISSUE THREAD

**Issue by saichint** [ansible/ansible#30310]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
components which use "show running-config all" etc.

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel 54cd641df4) last updated 2017/07/19 18:06:44 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/p2/Projects/nxos_ansible/p1_ansible/lib/ansible
  executable location = /root/p2/Projects/nxos_ansible/p1_ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### OS / ENVIRONMENT

Ansible Server : Ubuntu 14.04
Device: NX-OSv9K running 7.0(3)I2(5)
##### SUMMARY

If the transport is CLI and during the get_config() methods, if we pass flags like "all" etc. the current code in nxos.py is just ignoring the flags and it is making the flags as always []. This is causing lots of issues in various modules.

The issue came from this PR: https://github.com/ansible/ansible/pull/29839
This is the culprit in nxos.py file:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/nxos.py#L118
which is 

flags = [] if flags is None else []

So basically flags is [] irrespective of the value passed.

It should be same as nxapi call:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/nxos.py#L306
which is

flags = [] if flags is None else flags

nxapi is working properly because the code is correct here.

This is a MUST FIX

##### STEPS TO REPRODUCE

Just run vxlan_vtep playbook with transport as cli and the idempotent tests will fail

```yaml

```

##### ACTUAL RESULTS

```

```

**Comment by jborean93** [comment]
cc @GGabriele @jedelman8 @mikewiebe @privateip @rahushen @rcarrillocruz @trishnaguha
